stacker/hooks/keypair.py

```diff
@@ -46,62 +69,97 @@ def ensure_keypair_exists(provider, context, **kwargs):
         }
 
     logger.info("keypair: \"%s\" not found", keypair_name)
-    create_or_upload = raw_input(
-        "import or create keypair \"%s\"? (import/create/Cancel) " % (
-            keypair_name,
-        ),
-    )
-    if create_or_upload == "import":
-        path = raw_input("path to keypair file: ")
-        full_path = utils.full_path(path)
-        if not os.path.exists(full_path):
-            logger.error("Failed to find keypair at path: %s", full_path)
+    if ssm_parameter_name:
```

Comment:
Can you pull this out into a separate function?  It seems like a reasonable amount of code that could live in its own function.